test/streamingReceiver.spec.ts

```diff
@@ -247,4 +249,166 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
 
     await receiveListener.stop();
   });
+
+  it("Abandoned message is retained in the Queue with incremented deliveryCount. After 10 times, you can only get it from the dead letter queue.", async function(): Promise<
+    void
+  > {
+    await queueClient.sendBatch(testMessages);
+
+    let peekedMsgs = await queueClient.peek(2);
+    should.equal(peekedMsgs.length, 2);
+    should.equal(peekedMsgs[0].deliveryCount, 0);
+    should.equal(peekedMsgs[1].deliveryCount, 0);
+    let checkDeliveryCount0 = 0;
+    let checkDeliveryCount1 = 0;
+
+    const receiveListener = await queueClient.receive(
+      (msg: ServiceBusMessage) => {
+        if (msg.messageId === testMessages[0].messageId) {
+          should.equal(msg.deliveryCount, checkDeliveryCount0);
+          checkDeliveryCount0++;
+        } else if (msg.messageId === testMessages[1].messageId) {
+          should.equal(msg.deliveryCount, checkDeliveryCount1);
+          checkDeliveryCount1++;
+        }
+        msg.abandon();
+        return Promise.resolve();
+      },
+      (err: Error) => {
+        should.not.exist(err);
+      },
+      { autoComplete: false }
+    );
+
+    await delay(4000);
+
+    await receiveListener.stop();
+
+    should.equal(checkDeliveryCount0, maxDeliveryCount);
+    should.equal(checkDeliveryCount1, maxDeliveryCount);
+
+    peekedMsgs = await queueClient.peek(2);
+    should.equal(peekedMsgs.length, 0); // No messages in the queue
+
+    const deadLetterQueuePath = Namespace.getDeadLetterQueuePathForQueue(queueClient.name);
+    const deadletterQueueClient = namespace.createQueueClient(deadLetterQueuePath);
+
+    await testPeekMsgsLength(deadletterQueueClient, 2); // Two messages in the DLQ
```

Comment:
Since you receiving 2 messages from the deadletter queue, this check can be skipped.